cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowActionSpecification.java

```diff
@@ -16,24 +16,21 @@
 
 package co.cask.cdap.api.workflow;
 
-import co.cask.cdap.api.builder.Creator;
-import co.cask.cdap.api.builder.DescriptionSetter;
-import co.cask.cdap.api.builder.NameSetter;
-import co.cask.cdap.api.builder.OptionsSetter;
+import co.cask.cdap.api.ProgramSpecification;
 import co.cask.cdap.api.common.PropertyProvider;
-import co.cask.cdap.api.mapreduce.MapReduceSpecification;
-import co.cask.cdap.internal.builder.BaseBuilder;
-import co.cask.cdap.internal.builder.SimpleDescriptionSetter;
-import co.cask.cdap.internal.builder.SimpleNameSetter;
+import co.cask.cdap.api.dataset.Dataset;
 import co.cask.cdap.internal.workflow.DefaultWorkflowActionSpecification;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableSet;
 import com.google.common.collect.Maps;
 
 import java.util.Map;
+import java.util.Set;
 
 /**
  * Specification for a {@link WorkflowAction}.
  */
-public interface WorkflowActionSpecification extends PropertyProvider {
+public interface WorkflowActionSpecification extends ProgramSpecification, PropertyProvider {
```

Comment:
Not sure why I added it, as the interface already specifies the getClassName, getName and getDescription methods.  Would it be better to delete the "extend ProgramSpecification" or the 3 method declarations?